bundles/org.openhab.binding.sunsa/src/main/java/org/openhab/binding/sunsa/internal/device/SunsaDeviceHandler.java

```diff
@@ -0,0 +1,188 @@
+        // we set this upfront to reliably check status updates in unit tests.
+        updateStatus(ThingStatus.UNKNOWN);
+
+        // Example for background initialization:
+        scheduler.execute(() -> {
+            if (isConnected()) {
+                updateStatus(ThingStatus.ONLINE);
+            } // getState already updates the status with more detail if it thing is not reachable
+        });
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        if (command instanceof RefreshType) {
+            updateState();
+            return;
+        }
+
+        switch (channelUID.getId()) {
+            case CHANNEL_CONFIGURABLE_POSITION: {
+                if (command instanceof StopMoveType) {
+                    // We don't support this
+                    LOGGER.warn("Received {} command which is not supported and is explicitly ignored.", command);
+                    return;
+                }
+
+                final int rawPosition = getPositionAdapter(channelUID.getId())
+                        .getRawPosition(getPositionInterpolationFactor(command));
+                LOGGER.debug("Attempting update to raw position to: {}", rawPosition);
+                updatePosition(rawPosition);
+                break;
+            }
+
+            case CHANNEL_RAW_POSITION: {
+                final int rawPosition = ((DecimalType) command).intValue();
+                LOGGER.debug("Attempting update to raw position to: {}", rawPosition);
+                updatePosition(rawPosition);
+                break;
+            }
+
+            case CHANNEL_BATTERY_LEVEL: {
+                updateState();
+            }
+        }
+    }
+
+    public PositionAdapter getPositionAdapter(final String channelID) {
+        switch (channelID) {
+            case CHANNEL_CONFIGURABLE_POSITION:
+                return PositionAdapters.configurablePositionAdapter(config.configurablePositionClosed,
+                        config.configurablePositionOpen);
+            default:
+                throw new IllegalArgumentException("Unknown channel ID: " + channelID);
+        }
+    }
+
+    /**
+     * Return a value between [0.0, 1.0] for the value of the command.
+     */
+    private float getPositionInterpolationFactor(final Command command) {
+        final float rawValue;
+        if (command instanceof PercentType) {
+            rawValue = ((PercentType) command).floatValue();
+        } else if (command instanceof UpDownType) {
+            rawValue = ((UpDownType) command).as(PercentType.class).floatValue();
+        } else if (command instanceof OpenClosedType) {
+            rawValue = ((OpenClosedType) command).as(PercentType.class).floatValue();
+        } else if (command instanceof DecimalType) {
+            rawValue = ((DecimalType) command).floatValue();
+        } else {
+            throw new IllegalArgumentException("Unknown command: " + command);
```

Comment:
What happens if this Exception is thrown? It doesn't looks like it is handled upstream so it causses a thread crash? Similar for the other IllegalArgumentException